lib/query/woqlQuery.js

```diff
@@ -18,6 +18,57 @@ WOQLQuery.prototype.read_object = function(IRI, OutputVar, Format) {
     return this.wform(Format)
 }
 
+WOQLQuery.prototype.read_document = function(IRI, OutputVar, Format) {
+    //if (IRI && IRI == 'args') return ['document_uri', 'document']
+    if (this.cursor['@type']) this.wrapCursorWithAnd()
+    this.cursor['@type'] = 'ReadDocument'
+    this.cursor['identifier'] = this.cleanNodeValue(IRI)
+    this.cursor['document'] = this.expandValueVariable(OutputVar)
+    return this.wform(Format)
+}
+
+WOQLQuery.prototype.insert_document = function(docjson, IRI) {
+    //if (IRI && IRI == 'args') return ['document_uri', 'document']
+    if (this.cursor['@type']) this.wrapCursorWithAnd()
+    this.cursor['@type'] = 'InsertDocument'
+    if(typeof IRI !== 'undefined') this.cursor['identifier'] = this.cleanNodeValue(IRI)
+
+    let doc;
```

Comment:
Instead of declaring `doc` here, can you just do the assignment to `this.cursor['document']` directly in each block of the `if`/`else`?

Reply:
Ohh didn't look it that way